components/engine_rocks/src/engine.rs

```diff
@@ -105,17 +106,24 @@ impl Iterable for RocksEngine {
 }
 
 impl Peekable for RocksEngine {
-    fn get_opt(&self, opts: &ReadOptions, key: &[u8]) -> Result<Option<Vec<u8>>> {
+    type DBVector = RocksDBVector;
+
+    fn get_value_opt(&self, opts: &ReadOptions, key: &[u8]) -> Result<Option<RocksDBVector>> {
         let opt: RocksReadOptions = opts.into();
         let v = self.0.get_opt(key, &opt.into_raw())?;
-        Ok(v.map(|v| v.to_vec()))
+        Ok(v.map(RocksDBVector::from_raw))
     }
 
-    fn get_cf_opt(&self, opts: &ReadOptions, cf: &str, key: &[u8]) -> Result<Option<Vec<u8>>> {
+    fn get_value_cf_opt(
+        &self,
+        opts: &ReadOptions,
+        cf: &str,
+        key: &[u8],
+    ) -> Result<Option<RocksDBVector>> {
         let opt: RocksReadOptions = opts.into();
         let handle = get_cf_handle(&self.0, cf)?;
         let v = self.0.get_cf_opt(handle, key, &opt.into_raw())?;
-        Ok(v.map(|v| v.to_vec()))
+        Ok(v.map(RocksDBVector::from_raw))
```

Comment:
can we return `self.0.get_cf_opt(handle, key, &opt.into_raw())` directly?

Reply:
No, because RocksDBVector is not the same as the DBVector from the rocks crate and can't trivially be made to be the same type.